src/libstd/fs.rs

```diff
@@ -1771,12 +1771,29 @@ mod tests {
     }
 
     #[test]
+    fn file_create_new_path_already_exists_error() {
+        let tmpdir = tmpdir();
+        let file = &tmpdir.join("file_create_new_exists_file");
+        check!(fs::File::create(file));
+        let e = fs::OpenOptions::new().write(true).create_new(true).open(file).unwrap_err();
+        assert_eq!(e.kind(), ErrorKind::AlreadyExists);
+        let dir = &tmpdir.join("file_create_new_exists_dir");
+        check!(fs::create_dir(dir));
+        let e = fs::OpenOptions::new().write(true).create_new(true).open(dir).unwrap_err();
```

Comment:
Apparently on Windows an this (creating a file when e directory exists) results in `ErrorKind::PermissionDenied`. Should I just not test this or make it conditional on `!windows`?